core/src/main/java/org/mule/runtime/core/routing/AsynchronousUntilSuccessfulProcessingStrategy.java

```diff
@@ -72,19 +69,13 @@ public void initialise() throws InitialisationException {
 
   @Override
   public void start() {
-    final String threadPrefix =
-        String.format("%s%s.%s", ThreadNameHelper.getPrefix(getUntilSuccessfulConfiguration().getMuleContext()),
-                      getUntilSuccessfulConfiguration().getFlowConstruct().getName(), "until-successful");
-    pool = getUntilSuccessfulConfiguration().getThreadingProfile().createPool(threadPrefix);
-    scheduledRetriesPool = getUntilSuccessfulConfiguration().createScheduledRetriesPool(threadPrefix);
+    pool = muleContext.getSchedulerService().ioScheduler();
```

Comment:
Would it be a good idea to allow a name to be passed when a pool is received, so any logging or error that occur can be associated back to where pool is used.  Or is there another mechanism to do this?